marlowe-dashboard-client/src/Play/State.purs

```diff
@@ -122,11 +121,17 @@ handleAction (TemplateAction (Template.SetTemplate template)) = do
   when (mCurrentTemplate /= Just template) $ assign (_playState <<< _templateState) $ Template.mkInitialState template
   handleAction $ SetScreen TemplateScreen
 
-handleAction (TemplateAction Template.ToggleTemplateLibraryCard) = handleAction $ ToggleCard TemplateLibraryCard
+handleAction (TemplateAction Template.OpenTemplateLibraryCard) = handleAction $ OpenCard TemplateLibraryCard
 
-handleAction (TemplateAction (Template.ToggleCreateWalletCard tokenName)) = handleAction $ ToggleCard $ CreateWalletCard $ Just tokenName
+handleAction (TemplateAction (Template.OpenCreateWalletCard tokenName)) = handleAction $ OpenCard $ CreateWalletCard $ Just tokenName
 
-handleAction (TemplateAction Template.ToggleSetupConfirmationCard) = handleAction $ ToggleCard ContractSetupConfirmationCard
+handleAction (TemplateAction Template.OpenSetupConfirmationCard) = handleAction $ OpenCard ContractSetupConfirmationCard
+
+handleAction (TemplateAction Template.CloseSetupConfirmationCard) = do
+  mCards <- peruse (_playState <<< _cards)
+  for_ mCards \cards -> case lastOf traversed cards of
```

Comment:
If we are using lenses for this, we could join them in the previous call. 

```purescript
handleAction (TemplateAction Template.CloseSetupConfirmationCard) = do
  state <- get
  case lastOf (_playState <<< _cards <<< traversed) state of
    Just ContractSetupConfirmationCard -> handleAction CloseCard
    _ -> pure unit
```

In any case, why are we checking for the last card instead of just closing?

Reply:
we don't really need to check, but my thinking was that we wouldn't give the `Template` module the power to close any card, just in case :man_shrugging: